src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -370,12 +362,17 @@ public Uri GetProxy(Uri uri)
                 }
 
                 // We did not find match on bypass list.
-                return (uri.Scheme == UriScheme.Https || uri.Scheme == UriScheme.Wss) ? _secureProxyUri : _insecureProxyUri;
+                return IsSecureUri(uri) ? _secureProxyUri : _insecureProxyUri;
             }
 
             return null;
         }
 
+        private static bool IsSecureUri(Uri uri)
```

Comment:
I'm pretty sure we already have a helper that does this somewhere in the Uri or Http code.

Reply:
I'll take a look and see if I can find it; this method is a refactoring of an existing snip of code that wasn't calling any method.

Reply:
I can't find one. :/

Reply:
>I can't find one. :/

ok.  Let's  not block the PR on that. But I'm pretty sure we have this somewhere because it is a common thing to do to return bool if Uri is a secure HTTPS or WSS scheme.  If I find it,  then we can do a follow up PR in the future.